rain_api_core/aws_util.py

```diff
@@ -148,13 +148,16 @@ def get_role_creds(user_id: str='', in_region: bool=False):
     if not user_id:
         user_id = 'unauthenticated'
 
-    if in_region:
-        download_role_arn = os.getenv('EGRESS_APP_DOWNLOAD_ROLE_INREGION_ARN')
-    else:
-        download_role_arn = os.getenv('EGRESS_APP_DOWNLOAD_ROLE_ARN')
+    if user_id not in role_creds_cache:
```

Comment:
I think we need to keep user-A's Internal and External roles seperate.

Reply:
Good point.

Reply:
Updated.